ui/marketplace/ContractListButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { Link, Tooltip, Skeleton } from '@chakra-ui/react';
+import React from 'react';
+import type { MouseEvent } from 'react';
+
+import config from 'configs/app';
+import type { IconName } from 'ui/shared/IconSvg';
+import IconSvg from 'ui/shared/IconSvg';
+
+export enum ContractListButtonVariants {
+  ALL_CONTRACTS = 'all contracts',
+  VERIFIED_CONTRACTS = 'verified contracts',
+}
+
+const values = {
+  [ContractListButtonVariants.ALL_CONTRACTS]: {
+    icon: 'contracts' as IconName,
```

Comment:
Can we avoid type coercion here? TypeScript won't complain if I misspell the icon name.

Reply:
`as const` should do the trick